app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/Session.kt

```diff
@@ -0,0 +1,162 @@
+    val roomIndex: Int = 0,
+    val speakers: List<String> = emptyList(),
+    val track: String = "",
+    val type: String = "",
+    val language: String = "",
+    val slug: String = "",
+    val recordingLicense: String = "",
+    val recordingOptOut: Boolean = false,
+    val highlight: Boolean = false,
+    val hasAlarm: Boolean = false,
+
+    val changedTitle: Boolean = false,
+    val changedSubtitle: Boolean = false,
+    val changedRoomName: Boolean = false,
+    val changedDayIndex: Boolean = false,
+    val changedStartTime: Boolean = false,
+    val changedDuration: Boolean = false,
+    val changedSpeakers: Boolean = false,
+    val changedLanguage: Boolean = false,
+    val changedRecordingOptOut: Boolean = false,
+    val changedTrack: Boolean = false,
+
+    val changedIsNew: Boolean = false,
+    val changedIsCanceled: Boolean = false,
+) {
+
+    override fun equals(other: Any?) =
+        other is Session && EssentialSession(this) == EssentialSession(other)
+
+    override fun hashCode() = EssentialSession(this)
+        .hashCode()
+
+    override fun toString() = EssentialSession(this)
+        .toString()
+        .replaceFirst("EssentialSession", "Session")
+
+    /**
+     * Returns a moment based on the start time milliseconds.
+     *
+     * Don't use in [Conference.ofSessions] as long as [relStartTime] is supported.
+     * See: [5a402](https://github.com/EventFahrplan/EventFahrplan/commit/5a4022b00434700274a824cc63f6d54a18b06fac)
+     */
+    val startsAt: Moment
+        get() {
+            require(dateUTC > 0) { "Field 'dateUTC' must be more than 0." }
+            return Moment.ofEpochMilli(dateUTC)
+        }
+
+    /**
+     * Returns a moment based on summing up the start time milliseconds and the duration.
+     */
+    val endsAt: Moment
+        get() = Moment.ofEpochMilli(dateUTC + duration.toLong() * MILLISECONDS_OF_ONE_MINUTE)
+
+    val isChanged: Boolean
+        get() = changedTitle ||
+                changedSubtitle ||
+                changedRoomName ||
+                changedDayIndex ||
+                changedStartTime ||
+                changedDuration ||
+                changedSpeakers ||
+                changedLanguage ||
+                changedRecordingOptOut ||
+                changedTrack
+
+    /**
+     * Returns a new session with [changedIsCanceled] set to `true`
+     * and all other change flags set to `false`.
+     */
+    fun cancel() = copy(
+        changedIsCanceled = true,
+        changedIsNew = false,
+        changedTitle = false,
+        changedSubtitle = false,
+        changedRoomName = false,
+        changedDayIndex = false,
+        changedStartTime = false,
+        changedDuration = false,
+        changedSpeakers = false,
+        changedLanguage = false,
+        changedRecordingOptOut = false,
+        changedTrack = false,
+    )
+
+}
+
+private data class EssentialSession(
```

Comment:
For future consideration: This is a good indicator that `Session` is doing too much and should be split apart.

The `changed*` properties are rarely necessary and could live in a separate class

```kotlin
data class ChangedSession(
    val session: Session,

    val changedTitle: Boolean = false,
    val changedSubtitle: Boolean = false,
    val changedRoomName: Boolean = false,
    val changedDayIndex: Boolean = false,
    val changedStartTime: Boolean = false,
    val changedDuration: Boolean = false,
    val changedSpeakers: Boolean = false,
    val changedLanguage: Boolean = false,
    val changedRecordingOptOut: Boolean = false,
    val changedTrack: Boolean = false,
    val changedIsNew: Boolean = false,
    val changedIsCanceled: Boolean = false,
)
```

Reply:
:+1: Agreed. Very much. I noted it down for a follow-up refactoring.